src/emitMixin.ts

```diff
@@ -4,9 +4,9 @@ export const attachEmitListener = () => {
   return {
     beforeCreate() {
       let events: Record<string, unknown[]> = {}
-      this.__emitted = events
-      const originalEmit = getCurrentInstance().emit
-      getCurrentInstance().emit = (event: string, ...args: unknown[]) => {
+      ;(this as any).__emitted = events
+      const originalEmit = getCurrentInstance()!.emit
```

Comment:
why do we need `!` here?

Reply:
`getCurrentInstance` returns this instance or null  (see https://github.com/vuejs/vue-next/blob/38f2d23a607cd7077da189ac274a3a0ad542cc1f/packages/runtime-core/src/component.ts#L400) So TS is not happy if we don't check if it's not null. As the mixin is only called by us, I think we're safe with just `!`.

Reply:
Ok!